megatron/core/inference/unified_memory.py

```diff
@@ -1,6 +1,7 @@
 # Copyright (c) 2025, NVIDIA CORPORATION. All rights reserved.
 
 import os
+import torch
```

Comment:
I'm putting this comment on an unrelated diff, because this is about the pre-existing code.

We do `cudaSetDevice` but we never change it back. This sounds bad. I think the following should be added before `return ptr;` inside `managed_malloc`:
```
if (device != cur && device >= 0) cudaSetDevice(cur);
```